Source/Device.generated.swift

```diff
@@ -1088,6 +1088,26 @@ public enum Device {
     public var has5gSupport: Bool {
       return isOneOf(Device.allDevicesWith5gSupport) || isOneOf(Device.allDevicesWith5gSupport.map(Device.simulator))
     }
+    
+    /// All devices that support single ESIM.
+    public static var allDevicesWithSingleEsimSupport: [Device] {
+      return [.iPhoneX, .iPhoneXS, .iPhoneXSMax, .iPhoneXR, .iPhone11, .iPhone11Pro, .iPhone11ProMax, .iPhoneSE2, .iPhone12, .iPhone12Mini, .iPhone12Pro, .iPhone12ProMax, .iPadAir3, .iPad7, .iPad8, .iPad9, .iPad10, .iPadAir4, .iPadAir5, .iPadMini5, .iPadMini6, .iPadPro12Inch3, .iPadPro12Inch4, .iPadPro11Inch3, .iPadPro12Inch5, .iPadPro11Inch4, .iPadPro12Inch6]
+    }
+
+    /// Returns whether or not the device has single ESIM support.
+    public var hasSingleEsimSupport: Bool {
+      return isOneOf(Device.allDevicesWithSingleEsimSupport) || isOneOf(Device.allDevicesWithSingleEsimSupport.map(Device.simulator))
+    }
+    
+    /// All devices that support dual ESIM.
+    public static var allDevicesWithDualEsimSupport: [Device] {
+      return [.iPhone13, .iPhone13Mini, .iPhone13Pro, .iPhone13ProMax, .iPhoneSE3, .iPhone14, .iPhone14Plus, .iPhone14Pro, .iPhone14ProMax, .iPhone15, .iPhone15Plus, .iPhone15Pro, .iPhone15ProMax]
```

Comment:
Sorry for my late reply. Why are the devices with dual eSim support not included in the single eSim list?

Reply:
There's an 11" iPad that is missing from the single sim list.  Also, new 2024 iPads aren't in the list.